File: mobile/mediflow_mobile/lib/models/appointment.dart
class Appointment {
  final int id;
  final int doctorId;
  final String doctorName;
  final String specialty;
  final DateTime appointmentDateTime;
  final String status;
  final String? appointmentNumber;
  final double fee;

  const Appointment({
    required this.id,
    required this.doctorId,
    required this.doctorName,
    required this.specialty,
    required this.appointmentDateTime,
    required this.status,
    this.appointmentNumber,
    required this.fee,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json['id'] ?? 0,
      doctorId: json['doctorId'] ?? 0,
      doctorName: json['doctorName'] ?? 'Doctor',
      specialty: json['specialty'] ?? 'General Medicine',
      appointmentDateTime: DateTime.tryParse(json['appointmentDateTime'] ?? '') ?? DateTime.now(),
      status: json['status'] ?? 'Pending',
      appointmentNumber: json['appointmentNumber'],
      fee: (json['fee'] as num?)?.toDouble() ?? 0.0,
    );
  }
}

File: frontend/src/types/appointment.ts
export type AppointmentStatus =
  | 'Pending'
  | 'PaymentSubmitted'
  | 'Confirmed'
  | 'Completed'
  | 'Cancelled'
  | 'NoShow';

export interface Specialty {
  id: number;
  name: string;
  description?: string;
  iconName?: string;
}

export interface DoctorSpecialty {
  doctorId: number;
  specialtyId: number;
  specialty?: Specialty;
}

export interface Doctor {
  id: number;
  userId: number;
  name?: string;
  fullName?: string;
  slmcNumber?: string;
  hospital?: string;
  consultationFee: number;
  experienceYears: number;
  rating: number;
  reviewCount: number;
  profileImageUrl?: string;
  profilePhoto?: string;
  isAvailableToday?: boolean;
  specialties?: string[];
  doctorSpecialties?: DoctorSpecialty[];
  qualifications?: string;
  subSpecialty?: string;
  hospitalClinic?: string;
  languages?: string;
  location?: string;
  mbbsUniversity?: string;
  phdUniversity?: string;
  otherQualifications?: string;
  certifications?: string;
  age?: number;
  registrationNumber?: string;
}

export interface Appointment {
  id: number;
  patientId: number;
  doctorId: number;
  appointmentDateTime: string;
  status: AppointmentStatus;
  appointmentNumber?: string;
  notes?: string;
  fee?: number;
  createdAt: string;
  updatedAt: string;
  doctor?: Doctor;
  hasRated?: boolean;
}

export interface BookAppointmentRequest {
  doctorId: number;
  appointmentDateTime: string;
  notes?: string;
}

export interface RateAppointmentPayload {
  rating: number;
  review?: string;
}

export interface AppointmentRatingDto {
  id: number;
  appointmentId: number;
  doctorId: number;
  patientId: number;
  rating: number;
  review?: string;
  createdAt: string;
}
